src/components/TablePopover/TableSearchPopover.tsx

```diff
@@ -12,36 +12,40 @@ import { useOnClickOutside } from "hooks";
 const { blue, gray } = palette;
 
 interface TableSearchPopoverProps {
-  value: string;
-  onChange: (search: string) => void;
-  onConfirm: () => void;
   "data-cy"?: string;
+  onConfirm: (search: string) => void;
   placeholder?: string;
+  value: string;
 }
 
 export const TableSearchPopover: React.FC<TableSearchPopoverProps> = ({
   "data-cy": dataCy,
-  onChange,
   onConfirm,
   placeholder,
   value,
 }) => {
+  const [input, setInput] = useState(value);
   const [active, setActive] = useState(false);
-  const iconColor = value === "" ? gray.dark2 : blue.light1;
+  const iconColor = value === "" ? gray.dark2 : blue.base;
 
   const buttonRef = useRef(null);
   const popoverRef = useRef(null);
 
   // Handle onClickOutside
   useOnClickOutside([buttonRef, popoverRef], () => setActive(false));
 
-  const closePopup = () => {
-    onConfirm();
+  // If the value from the URL has changed, update the input.
+  useEffect(() => {
+    setInput(value);
+  }, [value]);
```

Comment:
What case does this handle, navigating to a URL that already has filters applied?

Reply:
It's because of when you use the "Clear all filters" button 😞 Since the component manages an internal state, if I don't clear it here, you'll still see the filter in the input even after clicking "Clear all filters"